### PR TITLE
Update election fencing documentation

### LINKED ISSUE #3084: Strict fencing
**Related dev. issue:** https://github.com/tarantool/tarantool/issues/7110
**Product:** Tarantool
**Since:** Tarantool 2.11
**Root documents:**
- https://www.tarantool.io/en/doc/latest/reference/configuration/#confval-election_fencing_enabled
- https://www.tarantool.io/en/doc/latest/how-to/replication/repl_leader_elect/
- https://www.tarantool.io/en/doc/latest/book/replication/repl_leader_elect/
- https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_api/modules/cartridge.argparse

**Announce:** https://github.com/orgs/tarantool/discussions/8371#user-content-strict-fencing-in-raft

# Details

The old parameter, `election_fencing_enabled`, will be deprecated in 2.11.
The `election_fencing_mode` parameter appears in 2.10.1.

---

In `box.cfg` option `election_fencing_enabled` is deprecated in favor
of `election_fencing_mode`. `election_fencing_mode` can be set to one
of the following values:
'off' - fencing turned off (same as `election_fencing_enabled` set to
false before).
Connection death timeout is 4*replication_timeout for all nodes.

'soft' (default) - fencing turned on, but connection death timeout is
the same for leader and followers in replicaset. This is enough to
solve cluster being readonly and not being to elect a new leader in
some situations because of pre-vote.
Connection death timeout is 4*replication_timeout for all nodes.

'strict' - fencing turned on. In this mode leader tries its best to
resign leadership before new leader can be elected. This is achived
by halving death timeout on leader.
Connection death timeout is 4*replication_timeout for followers and
2*replication_timout for current leader.
Requested by @grafin in https://github.com/tarantool/tarantool/commit/64ae9a0802f4dc7469c27c170c5b4d72bc777d4a.

### PR DESCRIPTION
**Pull request by andreyaksenov**
Moved changes from the https://github.com/tarantool/doc/pull/3097 PR with some cosmetic updates.

Updated topics:
- https://docs.d.tarantool.io/en/doc/2.11/concepts/replication/repl_leader_elect/#leader-election-process
- https://docs.d.tarantool.io/en/doc/2.11/how-to/replication/repl_leader_elect/#repl-leader-elect-config
- https://docs.d.tarantool.io/en/doc/2.11/reference/configuration/#cfg-replication-election-fencing-mode
- https://docs.d.tarantool.io/en/doc/2.11/book/cartridge/cartridge_dev/#fencing

### CODE REVIEW

File: doc/how-to/replication/repl_leader_elect.rst [hunk 1]
```diff
@@ -20,12 +20,11 @@ Configuration
         election_timeout = <seconds>,
         replication_timeout = <seconds>,
         replication_synchro_quorum = <count>,
-        election_fencing_enabled = <boolean>
+        election_fencing_mode = <string>
     })
 
-*   ``election_mode`` -- specifies the role of a node in the leader election
```
Comment (grafin) [inline doc/how-to/replication/repl_leader_elect.rst:26]: Why this was removed? Seems unrelated to fencing option.

**Review by grafin — CHANGES_REQUESTED** [review]
Thanks for the patch! Only one comment from me

File: doc/how-to/replication/repl_leader_elect.rst (hunk 1 above)
Comment (andreyaksenov) [reply doc/how-to/replication/repl_leader_elect.rst:26]: Ooh, thanks - deleted by mistake. Thanks for catching that.

**Review by grafin — APPROVED** [review]
Thanks for quick fix! LGTM.
